src/Commands/stubs/Caddyfile

```diff
@@ -0,0 +1,38 @@
+{
+	{$CADDY_GLOBAL_OPTIONS}
+
+	frankenphp {
+		worker vendor/laravel/octane/bin/frankenphp-worker.php {$WORKER_COUNT}
+	}
+}
+
+{$SERVER_NAME} {
+	log {
+		# Redact the authorization query parameter that can be set by Mercure
+		format filter {
+			wrap console
+			fields {
+				uri query {
+					replace authorization REDACTED
+				}
+			}
+		}
+	}
+
+	route {
+		encode zstd gzip
+
+		# Mercure config is injected here
+		{$CADDY_SERVER_EXTRA_DIRECTIVES}
+
+		# If the requested file does not exist, pass to the worker script
+		@indexFiles file {
+			try_files public/{path} vendor/laravel/octane/bin/frankenphp-worker.php
```

Comment:
This is a security vulnerability.

The Caddyfile should have `root public/` to ensure the webroot is correctly set. While the `try_files` does rewrite with `public/` it's still a risk to have the webroot set incorrectly because a request path like `/../config/secrets.txt` could still escape out of `public/`. Caddy has special path handling to ensure cleaned paths don't escape the root, but if the root is not correctly set, then those protections aren't in place.

IIRC Laravel packages can publish files to userland directories. I think `frankenphp-worker.php` should probably be published to `public/`. That'll be necessary so that Caddy sees that file within the `public/` webroot instead of elsewhere. I haven't written Laravel packages in quite a while, so help wanted here to do this correctly. IIRC there's `public_path()` which can be used to get the correct directory to publish the file to.

Reply:
Unless I missed something, there are now vulnerabilities here.
We use `try_files` to check that the file exists in `public/`, and we use the hardcoded path to the worker script if it doesn't.

Try `curl http://127.0.0.1:8000/../package.json` for instance, and you get a 404, as expected.

Publishing the worker script in `public/` was indeed my first idea. FrankenPHP is designed to work like that. Ideally, the best option would even be to check if we're using FrankenPHP in worker mode or another server in `public/index.php`. But it's not how Octane is currently designed so I used this "hack" to make something a close as possible to the current design.

If the maintainers agree, Im totally ok to switch to publishing the script in `public/` (it would more fit FrankenPHP's design and will allow removing these hacks in the Caddyfile). WDYT @taylorotwell @driesvints?

Reply:
I refactored the code to publish the worker script. It's much simpler now and the vuln is gone!